hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotCache.java

```diff
@@ -181,7 +183,23 @@ public ReferenceCounted<OmSnapshot> get(UUID key) throws IOException {
       throw new OMException("SnapshotId: '" + key + "' not found, or the snapshot is no longer active.",
           OMException.ResultCodes.FILE_NOT_FOUND);
     }
-    return rcOmSnapshot;
+    return new UncheckedAutoCloseableSupplier<OmSnapshot>() {
+      private AtomicReference<Boolean> closed = new AtomicReference<>(false);
+      @Override
+      public OmSnapshot get() {
+        return rcOmSnapshot.get();
+      }
+
+      @Override
+      public void close() {
```

Comment:
The close() method safely decrements the reference count of the cached OmSnapshot, ensuring this happens at most once, even if close() is called multiple times. This helps manage resources and prevents accidental double release.